src/users/models.py

```diff
@@ -0,0 +1,63 @@
+from typing import TYPE_CHECKING, Union
+
+from django.conf import settings
+from django.contrib.auth.models import AbstractBaseUser, PermissionsMixin
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+
+from .managers import CustomUserManager
+
+if TYPE_CHECKING:
+    from django.forms import EmailField
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
+    email = models.EmailField(_("email address"), unique=True)
+    is_staff = models.BooleanField(default=False)
+    is_active = models.BooleanField(default=True)
+    date_joined = models.DateTimeField(default=timezone.now)
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = []
+
+    objects = CustomUserManager()
+
+    def __str__(self) -> "EmailField":
+        return self.email
+
+
+class Profile(models.Model):
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    first_name = models.CharField(max_length=30, blank=True, null=True)  # noqa: DJ001
+    last_name = models.CharField(max_length=30, blank=True, null=True)  # noqa: DJ001
+    birth_date = models.DateField(blank=True, null=True)
+    gender = models.CharField(max_length=10, choices=[("M", "Male"), ("F", "Female")], blank=True, null=True)  # noqa: DJ001
+    weight = models.FloatField(blank=True, null=True)
+    height = models.FloatField(blank=True, null=True)
+    avatar = models.ImageField(upload_to="avatars/", blank=True, null=True)
+    # TODO czy jak dodaje blank=True to dodawać też null=True, wiem, ze róznica jest w zapisie w bazie
+    #  blank - "" pusty str null w bazie NULL ale jak to sie ma do naszej apki, czy jest róznica
```

Comment:
blank jest do walidacji formularzy modelu (np. admin, ModelForm) baza to olewa https://docs.djangoproject.com/pl/5.0/ref/models/fields/#null

null to parametr używany w bazie. https://docs.djangoproject.com/pl/5.0/ref/models/fields/#null

dla stringowych pól rób `(null=False, default="", blank=True)` - czyli pole nie może mieć nulla, default do bazy leci pusty string, a formularze pozwalają na nieuzupełnienie pola.

Reply:
popraw wobec tej zasady `filrst_name`, `last_name`,